xdsl/builder.py

```diff
@@ -27,11 +27,23 @@ class Builder:
     Operations will be inserted in this block.
     """
 
-    insertion_point: Operation | None = field(default=None)
+    _insertion_point: Operation | None = field(default=None)
     """
     Operations will be inserted before this operation, or at the end of the block if None.
     """
 
+    @property
+    def insertion_point(self) -> Operation | None:
+        return self._insertion_point
+
+    @insertion_point.setter
+    def insertion_point(self, insertion_point: Operation | None):
+        if insertion_point is not None:
+            if insertion_point.parent is not self.block:
```

Comment:
What about deeper nestings? Or will you just plae a different insertion point, if you need to add something within a region in the block you are building?

Reply:
Deeper nesting should have a different builder